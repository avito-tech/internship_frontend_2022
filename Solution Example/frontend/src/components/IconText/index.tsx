import React, { FC } from 'react';
import { Space } from 'antd';

interface IconTextProps {
    icon: React.FC
    text: string
}

const IconText: FC<IconTextProps> = ({ icon, text }) => (
    <Space data-testid='iconText'>
        {React.createElement(icon)}
        {text}
    </Space>
);

export default IconText;
