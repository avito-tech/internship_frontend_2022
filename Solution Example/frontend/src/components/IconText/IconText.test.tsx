import React from 'react';
import { render } from '@testing-library/react';
import { CheckCircleOutlined } from '@ant-design/icons';
import IconText from './index';

describe('IconText', () => {
    test('should render correctly', () => {
      const tree = render(<IconText icon={CheckCircleOutlined} text='test' />);

      expect(tree).toMatchSnapshot();
    });
});
