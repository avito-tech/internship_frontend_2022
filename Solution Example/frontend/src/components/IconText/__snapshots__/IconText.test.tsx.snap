// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconText should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
        data-testid="iconText"
      >
        <div
          class="ant-space-item"
          style="margin-right: 8px;"
        >
          <span
            aria-label="check-circle"
            class="anticon anticon-check-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="check-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
              />
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
            </svg>
          </span>
        </div>
        <div
          class="ant-space-item"
        >
          test
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
      data-testid="iconText"
    >
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        <span
          aria-label="check-circle"
          class="anticon anticon-check-circle"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="check-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
            />
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
      </div>
      <div
        class="ant-space-item"
      >
        test
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
